pkgs/development/compilers/ghc/7.10.1.nix

```diff
@@ -28,7 +28,9 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  # We patch Cabal for GHCJS. See: https://github.com/haskell/cabal/issues/2454
   preConfigure = ''
+    sed -i 's/HcPkg.useSingleFileDb = .*/HcPkg.useSingleFileDb = False/' libraries/Cabal/Cabal/Distribution/Simple/GHCJS.hs
```

Comment:
@peti: Yes, unfortunately. The fix (https://github.com/haskell/cabal/commit/975fc46e0055a9f9b434d594569a18eaa0ef0320) didn't make it into Cabal 1.22.2.0 release.